### PR TITLE
i3424 video reserves app

### PR DESCRIPTION
**Pull request by kayiwa**
- initialize clean role for video reserves 
- add a role to install video reserves application
- add tests for the new role
Co-authored-by: Christina Chortaria <christinach@users.noreply.github.com>

### CODE REVIEW

**Comment by maxkadel** [comment]
Can we add the `.htaccess` from the application repository to this role? See https://github.com/PrincetonUniversityLibrary/video_reserves/blob/c72cd038b8f5d6155aebee75468a75bdbd834604/hrc/.htaccess

File: roles/ealapps/tasks/main.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@
     update: true
```
Comment (sandbergja) [inline roles/ealapps/tasks/main.yml:7]: Could we please also generate the hrc/admin/config.php file with the database credentials?

Here is the syntax of that file: https://github.com/PrincetonUniversityLibrary/video_reserves/blob/c72cd038b8f5d6155aebee75468a75bdbd834604/hrc/admin/config.php.tpl

Link to example tasks: https://github.com/pulibrary/princeton_ansible/blob/0abed277291c880aa9c6f4fb2349ac9e6e632586/roles/libwww/tasks/main.yml#L5-L19

File: roles/ealapps/tasks/main.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@
...
     force: true
```
Comment (sandbergja) [inline roles/ealapps/tasks/main.yml:8]: Could we add the .htaccess file at hrc/.htaccess as part of this role?

Here is what it should include: https://github.com/PrincetonUniversityLibrary/video_reserves/blob/c72cd038b8f5d6155aebee75468a75bdbd834604/hrc/.htaccess

File: inventory/all_projects/vra [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[vr_production]
```
Comment (sandbergja) [inline inventory/all_projects/vra:1]: This file name should match the name of the app (video_reserves)

File: group_vars/vr/staging.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+mysql_server: false
+
+mysql_host: "mysql-db-staging1.princeton.edu"
+
+mysql_root_password: "{{ vault_maria_mysql_root_password }}"
+mysql_databases:
+  - name: "vr_staging_db"
```
Comment (sandbergja) [inline group_vars/vr/staging.yml:8]: Rename the vars from vr_ to video_reserves?

File: inventory/by_environment/staging [hunk 5]
```diff
@@ -49,4 +49,5 @@ sftp_staging
 shxco_staging
```
Comment (sandbergja) [inline inventory/by_environment/staging:49]: Could you please add this app to the inventory for apache?

File: roles/video_reserves/vars/main.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+---
+# vars file for video_reserves
+vr_git_repo: "git@github.com:PrincetonUniversityLibrary/video_reserves.git"
+apache_doc_root: "/var/www/vr"
+server_name: "{{ ansible_hostname }}"
+app_db_host: "{{ db_host | default('localhost') }}"
+app_db_user: "{{ vr_db_user | default('vr_db_user') }}"
+app_db_password: "{{ vr_db_password | default('change_me') }}"
+app_db_port: "{{ vr_db_port | default('3306') }}"
```
Comment (sandbergja) [inline roles/video_reserves/vars/main.yml:9]: The code is currently hardcoded to the snakeoil cert: https://github.com/PrincetonUniversityLibrary/video_reserves/blob/0b5536ad637fc0b11189e342420b600a5c38f13b/hrc/vod/clip.php#L32  -- We'd like to create a real cert instead and update the code accordingly.

File: roles/video_reserves/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  min_ansible_version: "2.4"
+
+  platforms:
+    - name: Ubuntu
+      versions:
+        - bionic
+dependencies:
+  - role: "deploy_user"
+  - role: "php"
+  - role: "mysql"
```
Comment (sandbergja) [inline roles/video_reserves/meta/main.yml:19]: Could you please add a dependency to the capistrano role?

**Review by sandbergja — COMMENTED** [review]
Thanks, @kayiwa !

File: roles/ealapps/tasks/main.yml (hunk 1 above)
Comment (maxkadel) [reply roles/ealapps/tasks/main.yml]: @kayiwa I just ran the playbook on the staging server and the `admin/config.php` file has not been created.

Comment (maxkadel) [reply roles/ealapps/tasks/main.yml]: Also, this comment should be for the video_reserves project - is it intentional that this PR has changes to both video_reserves and ealapps?

File: roles/video_reserves/templates/config.tpl.j2 [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+// {{ ansible_managed | comment }}
+// mysql info
+//
+define ('HOST', '{{ app_db_host }}');
+define ('USER', '{{ app_db_user }}');
+define ('PASS', '{{ app_db_password }}');
+define ('DB', '{{ video_reserves_db_name }}');
+define ('APP_URL', 'https://{{ ansible_hostname }}.princeton.edu/hrc'); // without training slash
```
Comment (maxkadel) [inline roles/video_reserves/templates/config.tpl.j2:10]: training -> trailing

File: roles/video_reserves/meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  license: MIT
+
+  min_ansible_version: "2.4"
+
+  platforms:
+    - name: Ubuntu
+      versions:
+        - bionic
+dependencies:
+  - role: "deploy_user"
```
Comment (maxkadel) [inline roles/video_reserves/meta/main.yml:17]: Need to add composer role (updated to using composer-managed CAS)

File: roles/video_reserves/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  ansible.builtin.template:
+    src: config.tpl.j2
+    dest: "{{ apache_doc_root }}/hrc/admin/config.php"
+    owner: "{{ deploy_user }}"
+    group: www-data
+    mode: '0640'
+  when: running_on_server
+
+- name: video_reserves | Create certificate signing request (CSR) for self-signed certificate
+  community.crypto.openssl_csr_pipe:
```
Comment (maxkadel) [inline roles/video_reserves/tasks/main.yml:52]: Community module? Maybe there's a better one out there?

**Pull request merged**